app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,56 @@
+class PasswordResetsController < ApplicationController
+  before_action :get_user, :valid_user, :check_expiration, only: %i(edit update)
+
+  def new; end
+
+  def edit; end
+
+  def create
+    @user = User.find_by email: params[:password_reset][:email].downcase
+    if @user
+      @user.create_reset_digest
+      @user.send_password_reset_email
+      flash[:info] = t ".instruction"
+      redirect_to root_url
+    else
+      flash.now[:danger] = t ".not_found"
+      render :new
+    end
+  end
+
+  def update
+    if params[:user][:password].empty?
+      @user.errors.add(:password, t(".error"))
+      render :edit
+    elsif @user.update user_params
+      log_in @user
+      flash[:success] = t ".success"
+      redirect_to @user
+    else
+      render :edit
+    end
+  end
+
+  private
+
+  def user_params
+    params.require(:user).permit :password, :password_confirmation
+  end
+
+  def get_user
+    @user = User.find_by email: params[:email]
+  end
+
+  def valid_user
+    return if @user.activated && @user.authenticated?(:reset, params[:id])
+
+    redirect_to root_url
+  end
+
+  def check_expiration
```

Comment:
```suggestion
  def check_password_expired
```
Đặt tên method tương mình 1 tí e